src/pybind/mgr/volumes/module.py

```diff
@@ -361,18 +361,45 @@ class Module(orchestrator.OrchestratorClientMixin, MgrModule):
         # volume in the lifetime of this module instance.
     ]
 
+    MODULE_OPTIONS = [
+        {
+            'name': 'cloner_threads',
+            'type': 'int',
+            'default': 4,
+            'desc': 'Number of asynchronous cloner threads',
+        }
+    ]
+
     def __init__(self, *args, **kwargs):
+        self.inited = False
         super(Module, self).__init__(*args, **kwargs)
+        self.cloner_threads = None
+        self.config_notify()
+
         self.vc = VolumeClient(self)
         self.fs_export = FSExport(self)
         self.nfs = NFSCluster(self)
+        self.inited = True
 
     def __del__(self):
         self.vc.shutdown()
 
     def shutdown(self):
         self.vc.shutdown()
 
+    def config_notify(self):
+        """
+        This method is called whenever one of our config options is changed.
+        """
+        for opt in self.MODULE_OPTIONS:
+            setattr(self,
+                    opt['name'],  # type: ignore
+                    self.get_module_option(opt['name']))  # type: ignore
+            self.log.debug(' mgr option %s = %s',
+                           opt['name'], getattr(self, opt['name']))  # type: ignore
+            if self.inited:
```

Comment:
module could have initialized `self.vc` but not yet set `self.inited` -- maybe serialize via mutex?

Reply:
Done.